src/Components/Publishing/Display/Canvas/CanvasContainer.tsx

```diff
@@ -85,6 +95,12 @@ export class CanvasContainerComponent extends React.Component<CanvasContainerPro
       layout
     }
 
+    // TODO: For whatever the slideshow leads to a race condition when rendering
+    // app in Force. Defer execution and everything works fine.
+    if (!this.state.isMounted) {
+      return <div />
```

Comment:
~11 hours, give or take

Reply:
Are there ways to potentially _potentially_ highlight these edge cases in danger?

Reply:
In this instance not necessarily, as there's no reason _why_ this would be an issue. I think a lot of these things could resolve if the publishing code followed more of a single responsibility principle, which would be a good refactor once the deadline is passed. There are a lot of instances where one component does many things internally, including accepting dynamic `{children}` from unknown places, so its hard to track down where one things begins and another ends.

Reply:
@orta - In terms of potental Danger plugins though -- yeah! libraries that aren't supported by SSR should be flagged or explicitely warned against. [`react-sizeme`](https://github.com/ctrlplusb/react-sizeme#server-side-rendering) is one of them, which was used here, and [`react-slick`](https://github.com/akiran/react-slick/issues/274) is another. This issue will always be present when dealing with SSR code. I think the solution is more awareness, as its easy to build something in storybooks (no SSR) and have it break in force.  Which leads me to think... it would be nice if there was an SSR-aware version of storybooks.